### ISSUE TITLE
Addon failing to get directory content from Netflix

### ISSUE THREAD

**Issue by hbbernardo** [asciidisco/plugin.video.netflix#311]
*I'm submitting a ...*
  - [ ] bug report
  

## General infomration

I have just noted that. I am unable to browse the content (any of it) using the Netflix Addon.

It started an hour ago. I am trying to browse Continuing Watching, Netflix Originals, New Releases et al

I am only getting a busy/working icon. 

* [ ] Are you running the latest version?

Yes. I am running the very latest version

### Description

[Description of the bug or feature]

### Steps to Reproduce

1. [First Step] Open Netflix addon. 
2. [Second Step] Try to browse it.

* [ ] I´m using the Netflix Repo

### Operating System

* [ ] Linux (ARM)

#### Additional informatin on the environment

[Descripe your environment a bit more detailed, are you using LibreElec f.e.]

Tested on LibreELEC and OSMC both running Leia. 

## Debug log

https://pastebin.com/8Jkzc7ky

**Comment by Zepmann** [comment]
I can confirm. Get the same errors:
```
05:29:35.781 T:7116   ERROR: XFILE::CDirectory::GetDirectory - Error getting plugin://plugin.video.netflix/?action=video_list&type=continueWatching&video_list_id=033a89d7-b960-42ba-b9ef-756c8d885143_4765787X3XX1521174571585
05:29:35.781 T:7116   ERROR: CGUIMediaWindow::GetDirectory(plugin://plugin.video.netflix/?action=video_list&type=continueWatching&video_list_id=033a89d7-b960-42ba-b9ef-756c8d885143_4765787X3XX1521174571585) failed
```

Tested under latest Kodi nightly (64-bit) on Windows (non-UWP).

**Comment by rfrayer420** [comment]
100% exact same symptoms as hbbernardo. LibreElec as well.

**Comment by jasonpro4584** [comment]
same here libreelec the only thing that works is search then it will play. even export works then i can watch the exported library

**Comment by SCiunczyk** [comment]
I have this same symptoms as above.

**Comment by fumachu** [comment]
Having the same symptoms with Kodi nightly 20171104-a5436a4 under Linux (Intel) with latest add-on from repository.

Browsing stopped working some hours ago. I can still play using the search.

**Comment by putyourlitterhere** [comment]
Same error message and behavior as described by @hbbernardo 
I Can not even use search function

**Comment by rochy-roch** [comment]
Yup same problem here.

**Comment by petvoigt** [comment]
I have exactly the same error situation on my raspi 3b: no directory content but working search:
LibreELEC (Milhouse): devel-20180315231318-#0315-gf291881 [Build #0315]

**Comment by fumachu** [comment]
Following [this blog post from securus](https://www.kodinerds.net/index.php/Thread/55612-PreRelease-Plugin-Netflix-Inputstream/?postID=452672#post452672) I was able to get browsing working again.

I changed the file `resources/lib/NetflixSession.py` in function `fetch_video_list` starting at line 1422 by changing the assignment to variable `paths`: 

- enclosed the parameter `list_id` in []-brackets (`list_id` -> `[list_id]`)
- added the new parameter `'reference'`

```        
$ diff NetflixSession_original.py NetflixSession_patched.py 
1422,1436c1422,1436
<             ['lists', list_id, {'from': list_from, 'to': list_to}, ['summary', 'title', 'synopsis', 'regularSynopsis', 'evidence', 'queue', 'episodeCount', 'info', 'maturity', 'runtime', 'seasonCount', 'releaseYear', 'userRating', 'numSeasonsLabel', 'bookmarkPosition', 'watched', 'delivery']],
<             ['lists', list_id, {'from': list_from, 'to': list_to}, 'cast', {'from': 0, 'to': 15}, ['id', 'name']],
<             ['lists', list_id, {'from': list_from, 'to': list_to}, 'cast', 'summary'],
<             ['lists', list_id, {'from': list_from, 'to': list_to}, 'genres', {'from': 0, 'to': 5}, ['id', 'name']],
<             ['lists', list_id, {'from': list_from, 'to': list_to}, 'genres', 'summary'],
<             ['lists', list_id, {'from': list_from, 'to': list_to}, 'tags', {'from': 0, 'to': 9}, ['id', 'name']],
<             ['lists', list_id, {'from': list_from, 'to': list_to}, 'tags', 'summary'],
<             ['lists', list_id, {'from': list_from, 'to': list_to}, ['creators', 'directors'], {'from': 0, 'to': 49}, ['id', 'name']],
<             ['lists', list_id, {'from': list_from, 'to': list_to}, ['creators', 'directors'], 'summary'],
<             ['lists', list_id, {'from': list_from, 'to': list_to}, 'bb2OGLogo', '_400x90', 'png'],
<             ['lists', list_id, {'from': list_from, 'to': list_to}, 'boxarts', '_342x192', 'jpg'],
<             ['lists', list_id, {'from': list_from, 'to': list_to}, 'boxarts', '_1280x720', 'jpg'],
<             ['lists', list_id, {'from': list_from, 'to': list_to}, 'storyarts', '_1632x873', 'jpg'],
<             ['lists', list_id, {'from': list_from, 'to': list_to}, 'interestingMoment', '_665x375', 'jpg'],
<             ['lists', list_id, {'from': list_from, 'to': list_to}, 'artWorkByType', 'BILLBOARD', '_1280x720', 'jpg']
---
>             ['lists', [list_id], {'from': list_from, 'to': list_to}, 'reference', ['summary', 'title', 'synopsis', 'regularSynopsis', 'evidence', 'queue', 'episodeCount', 'info', 'maturity', 'runtime', 'seasonCount', 'releaseYear', 'userRating', 'numSeasonsLabel', 'bookmarkPosition', 'watched', 'delivery']],
>             ['lists', [list_id], {'from': list_from, 'to': list_to}, 'reference', 'cast', {'from': 0, 'to': 15}, ['id', 'name']],
>             ['lists', [list_id], {'from': list_from, 'to': list_to}, 'reference', 'cast', 'summary'],
>             ['lists', [list_id], {'from': list_from, 'to': list_to}, 'reference', 'genres', {'from': 0, 'to': 5}, ['id', 'name']],
>             ['lists', [list_id], {'from': list_from, 'to': list_to}, 'reference', 'genres', 'summary'],
>             ['lists', [list_id], {'from': list_from, 'to': list_to}, 'reference', 'tags', {'from': 0, 'to': 9}, ['id', 'name']],
>             ['lists', [list_id], {'from': list_from, 'to': list_to}, 'reference', 'tags', 'summary'],
>             ['lists', [list_id], {'from': list_from, 'to': list_to}, 'reference', ['creators', 'directors'], {'from': 0, 'to': 49}, ['id', 'name']],
>             ['lists', [list_id], {'from': list_from, 'to': list_to}, 'reference', ['creators', 'directors'], 'summary'],
>             ['lists', [list_id], {'from': list_from, 'to': list_to}, 'reference', 'bb2OGLogo', '_400x90', 'png'],
>             ['lists', [list_id], {'from': list_from, 'to': list_to}, 'reference', 'boxarts', '_342x192', 'jpg'],
>             ['lists', [list_id], {'from': list_from, 'to': list_to}, 'reference', 'boxarts', '_1280x720', 'jpg'],
>             ['lists', [list_id], {'from': list_from, 'to': list_to}, 'reference', 'storyarts', '_1632x873', 'jpg'],
>             ['lists', [list_id], {'from': list_from, 'to': list_to}, 'reference', 'interestingMoment', '_665x375', 'jpg'],
>             ['lists', [list_id], {'from': list_from, 'to': list_to}, 'reference', 'artWorkByType', 'BILLBOARD', '_1280x720', 'jpg']

```

**Comment by zag2me** [comment]
Same issue here, thanks for the fix, will try it manually.

Can someone PR it?

**Comment by SqyD** [comment]
It seems there is a PR for this: https://github.com/asciidisco/plugin.video.netflix/pull/312/

**Comment by rochy-roch** [comment]
@fumachu
Thanks for the info, that saved me from using my TVs terrible app.

**Comment by Knut-M** [comment]
Here is a fully edited file which you have to replace in /storage/.kodi/addons/plugin.video.netflix/resources/lib
gl to everyone.

https://mega.nz/#!KUgnQAiK!cd4m7bU_R93eoI936JjFWkWh-3xICCS1zz7Dx2zFVH4

**Comment by hbbernardo** [comment]
@Knut-M 

Are you sure this is the patched version? 

I run a diff and there aren't any differences between the file you uploaded to the original.

**Comment by Knut-M** [comment]
Sry, my bad. I've changed the file :)

**Comment by Zepmann** [comment]
You can also get the patched file from [gismo112's pull request](https://github.com/gismo112/plugin.video.netflix/blob/master/resources/lib/NetflixSession.py).

**Comment by kartana** [comment]
Thanks for getting this fixed!

**Comment by hansworst24** [comment]
Yes it works again :) Only thing left remaining is updating the Repository, cause it's still at .7 version 😄

**Comment by SCiunczyk** [comment]
Big thanks !

**Comment by devil-strike** [comment]
Ok updated it but still not working, no episodes in directory's.

this on latest millhouse 9 builds.

**Comment by nicrame** [comment]
Thank You!!! For me it's working on newest LibreELEC (Milhouse): devel-20180324212711-#0324-g5c608c4 [Build #0324].

**Comment by sorin-pop** [comment]
I managed to fix this by copying over the corrected NetflixSession.py, and it worked for a while. But now, the same issue reoccurred, just like that. I had a look at the file above. It still contains the modifications I made. Anybody else experiencing the same?

**Comment by ccope** [comment]
I've noticed that sometimes it won't load the contents of the main directories until I switch back and forth between Netflix user profiles. Seems like a stale session or something.

**Comment by fragmental** [comment]
I was having this problem on Linux Mint 18, among other problems with Kodi, and I had to `sudo apt-get remove python-openssl` to fix some of the problems.  I ended up deleting all of my plugins/etc. and downgrading to 17 before trying this, so I can't confirm if this would have helped with the Netflix plugin.  Also, if you have windows then this probably doesn't affect you.  But it might help somebody.
